app/src/main/java/com/duckduckgo/app/entities/api/EntityListDownloader.kt

```diff
@@ -52,12 +52,12 @@ class EntityListDownloader @Inject constructor(
                 Timber.d("Updating entity list data from server")
                 val body = response.body()!!
 
-                var entities = emptyList<EntityListEntity>()
+                var entities = ArrayList<EntityListEntity>()
```

Comment:
A few suggestions:
1. `var entities` can become `val`
1. although equivalent, i prefer `mutableListOf<EntityListEntity>()` instead of `ArrayList<EntityListEntity>()`. I think it better signals intent that it's deliberately mutable
1. `domainName = it` shows a yellow warning in the IDE for me (implicit shadowed param `it`). could explicitly name it to avoid using `it` here?

         val entities = mutableListOf<EntityListEntity>()
         body.entries.forEach {
               val entity = it.key

               (it.value.properties + it.value.resources).forEach { domain ->
                    entities.add(EntityListEntity(entityName = entity, domainName = domain))
               }
          }

Reply:
Scope creep!!! 😜

But good suggestions. I've done the first two but left the last one out as it's not introduced here, all over the app and I don't mind it. Let's cover that as a separate conversation.

Reply:
a one-line change to make the file warning-free when you were already in there seems like a nice touch - thought that'd be right up your street. (and would have been quicker than this back and forth 😉)

but fair enough. it is out with the original PR intent.

Reply:
I'm just not sure I _agree_ with the warning. I'm sitting on the fence about whether we should conform or switch it off. I'll add it to next week's meeting as a point of style to agree on.

Reply:
Sure. For when the time comes: 

> In lambdas which are short and not nested, it's recommended to use the `it` convention instead of declaring the parameter explicitly. In nested lambdas with parameters, parameters should be always declared explicitly.

https://kotlinlang.org/docs/reference/coding-conventions.html